lib/internal/http2/core.js

```diff
@@ -2841,6 +2852,40 @@ class Http2Server extends NETServer {
   }
 }
 
+Http2Server.prototype[EventEmitter.captureRejectionSymbol] = function(
+  err, event, ...args) {
+
+  switch (event) {
+    case 'stream':
+      // TODO(mcollina): we might want to match this with what we do on
+      // the compat side.
+      const [stream] = args;
+      if (stream.sentHeaders) {
+        stream.destroy(err);
+      } else {
+        stream.respond({ [HTTP2_HEADER_STATUS]: 500 });
+        stream.end();
+      }
+      break;
+    case 'request':
+      const [, res] = args;
+      if (!res.headersSent && !res.finished) {
```

Comment:
I would probably prefer `writableFinished` over `finished` here since finished in http2 compat means something else.

Reply:
`res.finished` is close to being deprecated, use `writableEnded`?

Reply:
Note, the way this is currently written is probably wrong due to the above mentioned discrepancy in http2 compat (see https://github.com/nodejs/node/pull/24347).

Reply:
Have you got a quick example that would trigger the bug?

Reply:
Not 100% sure if the bug can apply here. But if you come here from within e.g. `'prefinish'` or `'finish'` handler then `res.finished` could be false even though `end()` has been called.

Reply:
Though the `headersSent` check probably guards against that. Maybe?

Reply:
I think we can leave this as is, and reassess if there are bug reports.